### PR TITLE
[SofaGraphComponent] Move scene checking ability into its own library

### PR DESCRIPTION
**Pull request by fredroy**
Scene checkers were located in the SofaGraphComponent (even if they were not components themselves....)
So they needed to be moved out of it.
The first idea was to move them to Sofa.Simulation as they related to the simulation itself (or rather its initialization).
But scene checking could need dependencies on components themselves (in this case, it needs the contact manager from Sofa.Component.Collision.Response.Contact, and APIVersion from Sofa.Component.SceneUtility)
So finally a library was created into ~~applications/plugins, even if it is not really a plugin in the sofa-way of thinking (aka a set of components)~~ Sofa/plugins because it is not in Component or framework, but it could be placed in Sofa/ directly (where GUI is as well)

As usual, a compat layer has been added to warn devs of the move.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/plugins/SceneChecking/src/SceneChecking/init.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <SceneChecking/init.h>
+
+namespace sofa::scenechecking
+{
+    
+extern "C" {
+    SOFA_EXPORT_DYNAMIC_LIBRARY void initExternalModule();
+    SOFA_EXPORT_DYNAMIC_LIBRARY const char* getModuleName();
```
Comment (alxbilger) [inline Sofa/plugins/SceneChecking/src/SceneChecking/init.cpp]: Following https://github.com/sofa-framework/sofa/pull/2948, you must also provide a version number

**Review by alxbilger — CHANGES_REQUESTED** [review]
Just a version number is requested

File: Sofa/plugins/SceneChecking/src/SceneChecking/init.cpp (hunk 1 above)
Comment (fredroy) [reply Sofa/plugins/SceneChecking/src/SceneChecking/init.cpp]: 👍

**Review by alxbilger — APPROVED** [review]

**Pull request merged**